### ISSUE TITLE
Methods in django model manager are not recognized

### ISSUE THREAD

**Issue by alphatownsman** [microsoft/pyright#5959]
**Describe the bug**
Methods in django model manager are not recognized. It happens for both user defined models and built-in models like ContentType.

**Code or Screenshots**

```python
from django.contrib.contenttypes.models import ContentType

ContentType.objects.get_for_model('')
```

```
$ pip install pyright django
$ pyright test.py
/Projects/test/test.py
  /Projects/test/test.py:3:21 - error: Cannot access member "get_for_model" for type "BaseManager[ContentType]"
    Member "get_for_model" is unknown (reportGeneralTypeIssues)
1 error, 0 warnings, 0 informations
```

**VS Code extension or command-line**
```
pyright --version
pyright 1.1.327
```
